src/frontend/src/lib/types/token.ts

```diff
@@ -66,3 +66,13 @@ interface TokenFinancialData {
 }
 
 export type TokenUi = Token & TokenFinancialData;
+
+export type TokenUiWithLinkedData = TokenUi & TokenFinancialData & RequiredTokenWithLinkedData;
+
+export interface TokenGroupUi {
+	header: TokenMetadata;
+	native: Network;
+	tokens: TokenUi[];
+}
+
+export type TokenUiOrGroupUi = TokenUi | TokenGroupUi;
```

Comment:
@AntonioVentilii-DFINITY do we have a TS utils for one or the other already? If not, I think I have that somewhere (for another PR)

Reply:
I remember that too, maybe I saw it in ic-js or gix-component... I will add it here and modify (I can do it in a separate  PR, so that it justifies why I am adding it)

Reply:
Sure - buts lets progress here, have 3 follow up prs depending on it for now.

Reply:
So, I will do it after in another PR and change it in this part? May you please ask Stefan to add it to the tasks? Thank you!!!

Reply:
Reported

Reply:
https://dfinity.atlassian.net/browse/OISY-141